### ISSUE TITLE
Crashing on openSUSE Tumbleweed with KDE

### ISSUE THREAD

**Issue by Woolba** [kozec/sc-controller#273]
sc-controller crashes on startup. Running openSUSE Tumbleweed w/ kernel 4.13.5 and KDE 5.11. Also using sc-controller 3.15-2.4 from the provided opensuse factory repository.

Terminal output from running sc-controller:

```
jordan@linux-h89d:~> sc-controller
(process:7912): Gtk-CRITICAL **: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 1131, in do_startup
    self.setup_widgets()
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 93, in setup_widgets
    self.builder.connect_signals(self)
ValueError: Pointer arguments are restricted to integers, capsules, and None. See: https://bugzilla.gnome.org/show_bug.cgi?id=683599
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 1101, in on_daemon_dead
    self.set_daemon_status("unknown", True)
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 1204, in set_daemon_status
    self.window.set_icon_from_file(icon)
AttributeError: 'App' object has no attribute 'window'
jordan@linux-h89d:~> pidfile /home/jordan/.config/scc/daemon.pid does not exist. Daemon not running?
```

Then again after running scc-daemon start:

```
jordan@linux-h89d:~> scc-daemon start
jordan@linux-h89d:~> sc-controller 
(process:8069): Gtk-CRITICAL **: gtk_icon_theme_get_for_screen: assertion 'GDK_IS_SCREEN (screen)' failed
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 1131, in do_startup
    self.setup_widgets()
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 93, in setup_widgets
    self.builder.connect_signals(self)
ValueError: Pointer arguments are restricted to integers, capsules, and None. See: https://bugzilla.gnome.org/show_bug.cgi?id=683599
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 720, in on_daemon_ccunt_changed
    self.profile_switchers[0].set_controller(None)
IndexError: list index out of range
Traceback (most recent call last):
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 868, in on_daemon_error
    if self.check():
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 179, in check
    ribar = self.show_error(msg)
  File "/usr/lib64/python2.7/site-packages/scc/gui/app.py", line 1069, in show_error
    if self.ribar is None:
AttributeError: 'App' object has no attribute 'ribar'
```

Let me know if you need additional information.

**Comment by kozec** [comment]
That's surprising amount of stuff going wrong in that log. Is there anything special on your setup, like out-of-date (or too new) GTK libs or entire thing running in Wayland? Are other GTK applications working? Is DISPLAY environment variable set? (run `echo $DISPLAY` for that last one)

**Comment by Woolba** [comment]
Nothing special that I know of (other than changing my r9 290's driver from radeon to amdgpu). Just did a fresh install last week since I built a new pc, so I haven't upgraded or downgraded any GTK libs outside of what is provided by Tumbleweed (libgtk-3-0 is version 3.22.24). I'm not running Wayland. Other GTK applications like gimp work. I also installed gnome clocks just to test, and that also works just fine.

Running `echo $DISPLAY` I get:
```
jordan@linux-h89d:~> echo $DISPLAY
:0

```

**Comment by kozec** [comment]
> Just did a fresh install last week since I built a new pc

Sounds like installing fresh Tumbleweed in VM should replicate it then. I'll try that.

**Comment by kozec** [comment]
So, right now it just looks like _something_ between GTK and OpenSUSE is broken, as that `self.builder.connect_signals` function simply behaves differently on Tumbleweed and differently on every other distro.

I'll check Gnome version of Tumbleweed tomorrow just in case it's somehow linked to DE.

**Comment by kozec** [comment]
@hezekiahc could you possibly list list of gtk-related packages you have  installed right now? Output of `rpm -qa | grep gtk` should be enough.

**Comment by hezekiahc** [comment]
@kozec I'll get that list to you in about 12 hours, once I get home. I'm fairly certain some of them are not needed, but failed to note which package installs had an effect on the error messages.

**Comment by hezekiahc** [comment]
@kozec   Here'e the output you requested, below. It appears there is only one GTK dev package. I checked my zipper history and found that I also installed several python gobject dev packages, after. It may have been a gobject dev package, and not a GTK dev package that got it working (sorry for the confusion). I've listed my gobject packages below, as well.

> rpm -qa | grep gtk
gtk3-immodule-inuktitut-3.22.24-3.1.x86_64
libcanberra-gtk2-module-0.30-14.3.x86_64
gtk4-lang-3.91.2-2.1.noarch
gettext-its-gtk4-3.91.2-2.1.x86_64
libwx_gtk2u_html-suse3-3.0.3-3.1.x86_64
libcanberra-gtk3-0-0.30-14.3.x86_64
libwebkit2gtk3-lang-2.18.0-3.1.noarch
gtk2-theming-engine-adwaita-3.22.3-1.2.x86_64
gtk3-branding-openSUSE-42.1-1.28.noarch
libcanberra-gtk2-module-32bit-0.30-14.3.x86_64
libwx_gtk2u_adv-suse-nostl3-3.0.3-3.1.x86_64
webkit2gtk-4_0-injected-bundles-2.18.0-3.1.x86_64
libcanberra-gtk0-32bit-0.30-14.3.x86_64
gtkspell-lang-2.0.16-19.1.noarch
gtk4-immodule-thai-3.91.2-2.1.x86_64
libwx_gtk2u_adv-suse3-3.0.3-3.1.x86_64
libqt5-qtstyleplugins-platformtheme-gtk2-5.0.0+git20170311-3.1.x86_64
libgtk-3-0-3.22.24-3.1.x86_64
at-spi2-atk-gtk2-2.26.0-1.1.x86_64
gtk2-immodule-vietnamese-2.24.31-3.2.x86_64
gtk4-immodule-vietnamese-3.91.2-2.1.x86_64
libwx_gtk2u_qa-suse-nostl3-3.0.3-3.1.x86_64
gtk3-data-3.22.24-3.1.noarch
libjavascriptcoregtk-4_0-18-2.18.0-3.1.x86_64
gtk4-data-3.91.2-2.1.noarch
gtk3-tools-3.22.24-3.1.x86_64
libgtkimageview0-1.6.4-21.9.x86_64
gtk3-immodule-amharic-3.22.24-3.1.x86_64
gtk2-immodule-amharic-2.24.31-3.2.x86_64
kde-gtk-config5-5.11.2-1.1.x86_64
kde-gtk-config5-gtk2-5.11.2-1.1.x86_64
gtk4-immodule-amharic-3.91.2-2.1.x86_64
libwx_gtk2u_gl-suse3-3.0.3-3.1.x86_64
gtk3-metatheme-breeze-5.11.2-1.1.noarch
gtk3-metatheme-adwaita-3.22.3-1.2.noarch
gtk-doc-lang-1.26-1.2.noarch
gtk3-immodule-vietnamese-3.22.24-3.1.x86_64
libcanberra-gtk-module-common-0.30-14.3.x86_64
libwx_gtk2u_core-suse3-3.0.3-3.1.x86_64
libgtkspell0-2.0.16-19.1.x86_64
gtk2-immodule-inuktitut-2.24.31-3.2.x86_64
python-gtk-2.24.0-16.2.x86_64
kde-gtk-config5-gtk3-5.11.2-1.1.x86_64
PackageKit-gtk3-module-1.1.7-3.1.x86_64
libgtk-4-0-3.91.2-2.1.x86_64
gtk4-immodule-inuktitut-3.91.2-2.1.x86_64
libwx_gtk2u_aui-suse3-3.0.3-3.1.x86_64
gtk2-metatheme-breeze-5.11.2-1.1.noarch
libcanberra-gtk3-module-0.30-14.3.x86_64
gtk-doc-1.26-1.2.noarch
libwx_gtk2u_qa-suse3-3.0.3-3.1.x86_64
gtk2-tools-2.24.31-3.2.x86_64
libgtk-2_0-0-2.24.31-3.2.x86_64
libgtk-2_0-0-32bit-2.24.31-3.2.x86_64
gtk3-devel-3.22.24-3.1.x86_64
gtk2-metatheme-adwaita-3.22.3-1.2.noarch
gtk4-branding-upstream-3.91.2-2.1.noarch
libwx_gtk2u_xrc-suse3-3.0.3-3.1.x86_64
libqt5-qtbase-platformtheme-gtk3-5.9.2-1.1.x86_64
gtk2-data-2.24.31-3.2.noarch
gtk2-engine-oxygen-32bit-1.4.6-4.2.x86_64
kde-gtk-config5-lang-5.11.2-1.1.noarch
gtk4-schema-3.91.2-2.1.noarch
gtk2-tools-32bit-2.24.31-3.2.x86_64
libgtkmm-2_4-1-2.24.5-2.2.x86_64
libwx_gtk2u_html-suse-nostl3-3.0.3-3.1.x86_64
gtk3-lang-3.22.24-3.1.noarch
libcanberra-gtk0-0.30-14.3.x86_64
libwebkit2gtk-4_0-37-2.18.0-3.1.x86_64
gtk3-immodule-thai-3.22.24-3.1.x86_64
gtk2-immodule-thai-2.24.31-3.2.x86_64
gtk2-branding-openSUSE-42.1-2.4.noarch
gtk4-tools-3.91.2-2.1.x86_64
libwx_gtk2u_core-suse-nostl3-3.0.3-3.1.x86_64

> rpm -qa | grep gobject
python2-gobject-Gdk-3.26.0-1.1.x86_64
libgobject-2_0-0-2.54.1-1.2.x86_64
gobject-introspection-devel-1.54.1-1.1.x86_64
libgobject-2_0-0-32bit-2.54.1-1.2.x86_64
python-gobject2-2.28.6-27.3.x86_64
python-gobject2-devel-2.28.6-27.3.x86_64
python3-gobject-Gdk-3.26.0-1.1.x86_64
python2-gobject-devel-3.26.0-1.1.x86_64
gobject-introspection-1.54.1-1.1.x86_64
python3-gobject-cairo-3.26.0-1.1.x86_64
python3-gobject-devel-3.26.0-1.1.x86_64
libcairo-gobject2-1.15.8-1.1.x86_64
python2-gobject-3.26.0-1.1.x86_64
python3-gobject-3.26.0-1.1.x86_64
python2-gobject-cairo-3.26.0-1.1.x86_64
python-gobject-common-devel-3.26.0-1.1.x86_64
